freqtrade/commands/build_config_commands.py

```diff
@@ -103,21 +103,32 @@ def ask_user_config() -> Dict[str, Any]:
             "message": "Please insert your display Currency (for reporting):",
             "default": 'USD',
         },
+        {
+            "type": "confirm",
+            "name": "trading_mode",
+            "message": "Do you want to trade Perpetual Swaps (perpetual futures)?",
+            "default": False,
+            "filter": lambda val: '"futures"' if val else '"spot"',
+        },
```

Comment:
I would inverse the sequence for this.

asking for the exchange first - and then determine if we support futures/margin with it or not - that way we can avoid asking for futures if it's not necessary.